static/extensions/MubiLop/numutils.js

```diff
@@ -0,0 +1,377 @@
+                defaultValue: 3
+              }
+            }
+          },
+          {
+            opcode: 'toExponential',
+            blockType: BlockType.REPORTER,
+            text: 'convert [NUMBER] to exponential notation with [DECIMALS] decimals',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 1234.5678
+              },
+              DECIMALS: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 2
+              }
+            }
+          },
+          {
+            opcode: 'formatCurrency',
+            blockType: BlockType.REPORTER,
+            text: 'format [NUMBER] as currency in [CURRENCY]',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 123.45
+              },
+              CURRENCY: {
+                type: ArgumentType.STRING,
+                menu: 'currencies',
+                defaultValue: 'USD'
+              }
+            }
+          },
+          {
+            opcode: 'formatPercentage',
+            blockType: BlockType.REPORTER,
+            text: 'format [NUMBER] as percentage with [DECIMALS] decimals',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 0.8547
+              },
+              DECIMALS: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 1
+              }
+            }
+          },
+          {
+            opcode: 'roundNumber',
+            blockType: BlockType.REPORTER,
+            text: 'round [NUMBER] to nearest [MULTIPLE]',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 127
+              },
+              MULTIPLE: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 10
+              }
+            }
+          },
+          {
+            opcode: 'isInteger',
+            blockType: BlockType.BOOLEAN,
+            text: 'is [NUMBER] an integer?',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 5
+              }
+            }
+          },
+          {
+            opcode: 'isFinite',
+            blockType: BlockType.BOOLEAN,
+            text: 'is [NUMBER] finite?',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 42
+              }
+            }
+          },
+          {
+            opcode: 'isNaN',
+            blockType: BlockType.BOOLEAN,
+            text: 'is [NUMBER] NaN?',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 'NaN'
+              }
+            }
+          },
+          {
+            opcode: 'parseNumber',
```

Comment:
numbers already get parsed by scratch automatically, giving this block no purpose